transformer_engine/common/util/padding.cu

```diff
@@ -0,0 +1,233 @@
+  const int tile_dim_n = THREADS_PER_WARP * desired_load_size / typeToSize(itype);
+
+  // Add tensors to kernel argument struct
+  MultiPaddingArgs kernel_args;
+  kernel_args.num_tensors = 0;
+  kernel_args.block_range[0] = 0;
+  for (size_t tensor_id = 0; tensor_id < input_list.size(); ++tensor_id) {
+    // Launch kernel if argument struct is full
+    if (kernel_args.num_tensors == kMaxTensorsPerKernel) {
+      TRANSFORMER_ENGINE_TYPE_SWITCH_ALL(
+          itype, InputType,
+          TRANSFORMER_ENGINE_TYPE_SWITCH_ALL(
+              otype, OutputType, constexpr int nvec_in = desired_load_size / sizeof(InputType);
+              constexpr int nvec_out = desired_store_size / sizeof(OutputType);
+              const int n_blocks = kernel_args.block_range[kernel_args.num_tensors];
+              multi_padding_kernel<nvec_in, nvec_out, fp32, InputType, OutputType>
+              <<<n_blocks, threads_per_block, 0, stream>>>(kernel_args););  // NOLINT(*)
+      );                                                                    // NOLINT(*)
+      kernel_args.num_tensors = 0;
+    }
+
+    // Calculate number of thread blocks needed for tensor
+    const int num_rows = input_list[tensor_id]->data.shape[0];
+    const int padded_num_rows = padded_num_rows_list[tensor_id];
+    const int row_length = input_list[tensor_id]->data.shape[1];
+    const int num_tiles_m = (padded_num_rows + tile_dim_m - 1) / tile_dim_m;
+    const int num_tiles_n = (row_length + tile_dim_n - 1) / tile_dim_n;
+    const int num_tiles = num_tiles_m * num_tiles_n;
+
+    // Add tensor to kernel argument struct
+    const int pos = kernel_args.num_tensors;
+    kernel_args.input_list[pos] = const_cast<void*>(input_list[tensor_id]->data.dptr);
+    kernel_args.output_list[pos] = output_list[tensor_id]->data.dptr;
+    kernel_args.num_rows_list[pos] = num_rows;
+    kernel_args.padded_num_rows_list[pos] = padded_num_rows;
+    kernel_args.row_length_list[pos] = row_length;
+    kernel_args.block_range[pos + 1] = kernel_args.block_range[pos] + num_tiles;
+    kernel_args.num_tensors++;
+  }
+
+  // Launch kernel
+  if (kernel_args.num_tensors > 0) {
+    TRANSFORMER_ENGINE_TYPE_SWITCH_ALL(
+        itype, InputType,
+        TRANSFORMER_ENGINE_TYPE_SWITCH_ALL(
+            otype, OutputType, constexpr int nvec_in = desired_load_size / sizeof(InputType);
+            constexpr int nvec_out = desired_store_size / sizeof(OutputType);
+            const int n_blocks = kernel_args.block_range[kernel_args.num_tensors];
+            multi_padding_kernel<nvec_in, nvec_out, fp32, InputType, OutputType>
+            <<<n_blocks, threads_per_block, 0, stream>>>(kernel_args););  // NOLINT(*)
+    );                                                                    // NOLINT(*)
+  }
+}
+
+}  // namespace transformer_engine
+
+void nvte_multi_padding(size_t num_tensors, const NVTETensor* input_list, NVTETensor* output_list,
```

Comment:
Hi, this kernel does both cast and padding. So I suggest renaming the kernel to make it more clear, for example `nvte_multi_cast_padding()`

Reply:
Ack.

Maybe I should simplify this kernel to remove cast. I hope this kernel can support all-type padding (including FP8). Its cast behavior is not fit to FP8.